larq/models.py

```diff
@@ -47,82 +64,286 @@ def _get_output_shape(layer):
+
+    @property
+    def fp_equivalent_memory(self):
+        return sum([p.fp_equivalent_memory for p in self.parameter_profiles])
+
+    def parameter_count(self, bitwidth=None, trainable=None):
+        count = 0
+        for p in self.parameter_profiles:
+            if bitwidth and (p.bitwidth != bitwidth):
+                continue
+            if (trainable is not None) and (p.trainable != trainable):
+                continue
+            count += p.count
+        return count
+
+    def op_count(self, precision=None, op_type=None, escape="?"):
+        if type(self._layer) in op_count_supported_layer_types:
+            count = 0
+            for op in self.op_profiles:
+                if precision and (op.precision != precision):
+                    continue
+                if op_type and (op.op_type != op_type):
+                    continue
+                count += op.n
+            return count
+        else:
+            return escape
+
+    def input_precision(self, default="-"):
+        try:
+            return self._layer.input_quantizer.precision
+        except:
+            return default
+
+    @property
+    def output_shape(self):
+        try:
+            return tuple(dim if dim else -1 for dim in self._layer.output_shape)
+        except AttributeError:
+            return "multiple"
+        except RuntimeError:  # output_shape unknown in Eager mode.
+            return "?"
+
+    @property
+    def output_pixels(self):
+        if len(self.output_shape) == 4:
+            return np.prod(self.output_shape[1:2])
+        elif len(self.output_shape) == 2:
+            return 1
+        else:
+            raise NotImplementedError()
+
+    @property
+    def unique_param_bidtwidths(self):
+        return sorted(np.unique([p.bitwidth for p in self.parameter_profiles]))
+
+    @property
+    def unique_op_precisions(self):
+        return sorted(np.unique([op.precision for op in self.op_profiles]))
+
+    def generate_table_row(self, table_config):
+        row = [self._layer.name, self.input_precision(), self.output_shape]
+        for i in table_config["param_bidtwidths"]:
+            n = self.parameter_count(i)
+            n = _format_table_entry(n, table_config["param_units"])
+            row.append(n)
+        row.append(_format_table_entry(self.memory, table_config["memory_units"]))
+        for i in table_config["mac_precisions"]:
+            n = self.op_count(i, "mac")
+            n = _format_table_entry(n, table_config["mac_units"])
+            row.append(n)
+
+        return row
+
+    def _quantized_weights(self):
+        try:
+            return self._layer.quantized_latent_weights
+        except:
+            return []
+
+    def _get_bitwidth(self, weight):
+        try:
+            quantizer = self._layer.quantizers[self._quantized_weights().index(weight)]
+            return quantizer.precision
+        except:
+            return 32
+
+
+class ModelProfile:
```

Comment:
Does it make sense to inherit from `LayerProfile` here?

Reply:
Currently I don't see a way to do this nicely. Eventually it would be good to support nested models.

Interesting to see how Keras will handle this: https://github.com/tensorflow/tensorflow/issues/28154

Reply:
Sorry for the confusion, I just ment that it would make sense to inherit `ModelProfile` from `LayerProfile` so you don't have to redefine `memory` and `fp_equivalent_memory`.

Reply:
Would be nice, but this requires to rename the `parameter_profiles` to `profiles` in `LayerProfile`, which I don't think is a good idea.